app/Controllers/authController.php

```diff
@@ -54,10 +54,10 @@ public function indexAction() {
 
 			if ($ok) {
 				Minz_Request::good(_t('feedback.conf.updated'),
-				                   array('c' => 'auth', 'a' => 'index'));
+								   array('c' => 'auth', 'a' => 'index'));
```

Comment:
I do not normally align my own code like that - this comes from @marienfressinaud - but I believe the intention is to get a stable alignment between all the lines. If I understand correctly, this change will only work for a precise number of spaces per tab, and not for other settings.
Maybe @marienfressinaud has a comment.
For the record, I normally use a single tab indentation in that case, but I do not really mind (as long as we keep tabs for general indentation 😃).

Reply:
I'll have a look. I personally prefer spaces :)

Reply:
I used to prefer tab for indentation and spaces for alignment (it was the case here), but today I use spaces only to avoid mixing both of them.

I think best option is to keep tab for indentation indeed (or we would have to change all the code!) but if we want to avoid mix with spaces, I would prefer something like that:

```php
Minz_Request::good(
	_t('feedback.conf.updated'),
	array('c' => 'auth', 'a' => 'index')
);
```

Reply:
That's the way I prefer it (all tabs, alignment optional and perhaps discouraged). Spaces for indentation just messes up ASCII art alignment in editors if you try to set a custom width and is therefore much worse than proper "mixing" (there's no mixing) of tabs and spaces. Everything looking the same everywhere is a bug, not a feature.

Anyway, that's my two cents. It's all fine though. ;-)

Reply:
Thats one way to look at it, another one is http://www.php-fig.org/psr/psr-2/

Reply:
I'd love to implement that one

Reply:
@kevinpapst just for fun, I've run phpcs with PSR2 and it's terrible. Every single file needs to be modified. Most of the errors are about spacing and can be fixed automatically, but some are really bad. Those are the one related to namespacing, class names for instance.
We have a long way to get there. Eventually, we will need to do that in order to get the latest php features by dropping support for older version. I can't wait to be there.

Reply:
Hm, not sure if I wanna go down that road. Don't get me wrong, I would totally prefer PSR-2 ... but has changing the complete codebase any benefits besides our personal feelings (especially when the other devs prefer tabs)? 
At least my IDE doesn't fu*k up all the whitespaces since I added the .editorconfig file. Its was a pain, as FreshRSS is the only project I work on that uses tabs.
Dropping support for older PHP version indeed has a lot of benefits, but a lot of downsides for users as well. So both topics should be carefully discussed, but likely in another place!?

Reply:
In the event of spaces, let the record show that I vote for 13 spaces per indentation. :angel: 

(That'll teach you how four spaces feel.)

Ftr, I've been called "worse than luacheck" for enforcing four spaces. :-P The project standard is the project standard; there's typically little reason to change that kind of thing.

Visually I typically prefer 2 spaces. In most editors there's the option to display tab as *n* spaces, but not to display, e.g., 4 spaces as 1 through 3 spaces. And in the rare case that there is, *unlike* proper indentation + spaces it messes up ASCII art, while proper tab indentation + spaces like in FreshRSS works fine with ASCII art. (Although I generally find ASCII art more distracting than helpful, regardless of indentation.) Using spaces is like using `\r\n`. :smiling_imp: 

Anyway, even if I preferred spaces I'd prefer to keep tabs because that's what currently happens to be there. Otherwise you'd just get a diff messing with git blame on every single line in every single file.

> At least my IDE doesn't fu*k up all the whitespaces since I added the .editorconfig file. Its was a pain, as FreshRSS is the only project I work on that uses tabs.

@kevinpapst Was auto-detect malfunctioning for some reason? I come across both plenty and it tends to work out fine.

Reply:
@Frenzie I literally haven't worked on a "tab" project for the last ten years, so I might have missed the correct IDE setting. Auto-detection wasn't working unless I added the .editorconfig. Problem could be in front of the screen, without doubt. 
I am voting against almost all code formatting rules for a longer time now, as everyone can easily reformat the code to his own liking. But that problem can't be fixed unless we see changes in the IDEs and probably the VCS as well (whitespaces changes are a nightmare not only for the history). The linked article is a great write down! Until then I would always vote for sticking to the project standards and ... ehm ... strongly against 13 spaces :-D